electron/src/renderer/App.jsx

```diff
@@ -0,0 +1,104 @@
+import { useState, useEffect, useContext } from 'react';
+import {
+  MemoryRouter as Router,
+  Switch,
+  Route,
+  Link,
+  Redirect,
+  useHistory,
+  useLocation,
+} from 'react-router-dom';
+import './App.css';
+import { realmApp, Authentication } from './realm';
+import Context from './components/Context';
+import LogIn from './components/LogIn';
+import SignUp from './components/SignUp';
+import Todo from './components/Todo';
+import Welcome from './components/Welcome';
+
+const PrivateRoute = (props) => {
+  const location = useLocation();
+  const { isLoggedIn } = useContext(Context);
+
+  return isLoggedIn ? (
+    <Route {...props} />
+  ) : (
+    <Redirect
+      to={{
+        pathname: '/',
+        state: { from: location },
+      }}
+    />
+  );
+};
+
+const App = () => {
+  const history = useHistory();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
+    console.log(realmApp.currentUser?.isLoggedIn);
+    setIsLoggedIn(!!realmApp.currentUser?.isLoggedIn);
+  }, [realmApp.currentUser?.isLoggedIn]);
+
+  useEffect(() => {
+    if (isLoggedIn) {
+      history.push('/todo');
+    }
+  }, [isLoggedIn]);
+
+  const onLogIn = async (username, password) => {
+    const res = await Authentication.logIn(username, password);
+    //checks if the response is an Error
+    if (res?.isError) {
+      return res;
+    } else {
+      setIsLoggedIn(true);
+      history.push('/todo');
+    }
+  };
+  const onSignUp = async (username, password) => {
+    const res = await Authentication.signUp(username, password);
+    //checks if the response is an Error
+    if (res?.isError) {
+      return res;
+    } else {
+      setIsLoggedIn(true);
+      history.push('/todo');
+    }
+  };
+
+  const onLogOut = () => {
+    Authentication.logOut();
+    setIsLoggedIn(false);
+  };
+
+  return (
+    <>
+      <Context.Provider value={{ onLogIn, onSignUp, isLoggedIn }}>
+        <div>
+          <Link to="/">
+            <h1>Realm Todo</h1>
+          </Link>
+          {isLoggedIn && <button onClick={onLogOut}>Log Out</button>}
+        </div>
+        <Switch>
+          <Route path="/" component={Welcome} exact />
+          <Route path="/log-in" component={LogIn} />
```

Comment:
this is fine for now, but we really should consolidate /log-in and /sign-up to be one single route